moveos/moveos-verifier/src/metadata.rs

```diff
@@ -459,13 +459,8 @@ impl<'a> ExtendedChecker<'a> {
     }
 }
 
-pub fn check_storage_context_struct_tag(struct_tag: String) -> bool {
-    let address = moveos_types::addresses::MOVEOS_STD_ADDRESS.to_string();
-    let module = StorageContext::module_identifier()
-        .as_ident_str()
-        .to_string();
-    let name = StorageContext::struct_identifier().to_string();
-    struct_tag == format!("{}::{}::{}", address, module, name)
+pub fn check_storage_context_struct_tag(struct_full_name: String) -> bool {
+    struct_full_name == StorageContext::struct_tag().to_canonical_string()
```

Comment:
The `to_canonical_string` does not contain the `0x` prefix, and the address is always a full string, not a short string. like `00000000000000000000000000000001`, maybe not work as you expected.

Reply:
At the source, I also used the same function:

```rust
let struct_module_address = module_view
                    .address_identifier_at(module_handle.address)
                    .to_canonical_string();
```

Because I tried here and couldn't get the short address.